content_scripts/vimium_frontend.coffee

```diff
@@ -96,6 +96,30 @@ settings =
 #
 frameId = Math.floor(Math.random()*999999999)
 
+# If an input grabs the focus before the user has interacted with the page, then grab it back.
+class GrabBackFocus extends Mode
+  constructor: ->
+    super
+      name: "grab-back-focus"
+      keydown: => @alwaysContinueBubbling => @exit()
+
+    @push
+      _name: "grab-back-focus-mousedown"
+      mousedown: => @alwaysContinueBubbling => @exit()
+
+    chrome.storage.sync.get "grabBackfocus", (items) =>
```

Comment:
It seems like bad and confusing practise to set this with our settings library but read it with the underlying mechanism.